### ISSUE TITLE
camelCase & snakeCase

### ISSUE THREAD

**Issue by shasharoman** [lodash/lodash#4708]
``` javascript
_.camelCase('x_y_z') => 'xYZ'
_.snakeCase('xYZ') => 'x_yz' // confusing

_.camelCase('x_yz') => 'xYz'
```

I think the above snakeCase's results are a bit confusing, what do you guys think, is this a bug or normal behavior.

**Comment by csvn** [comment]
In my opinion, this is working as intended. A couple of realistic scenarios:

```js
_.snakeCase('HTMLInputElement')
// 'html_input_element'
_.snaceCase('XmlHTTPRequest')
// 'xml_http_request'

_.camelCase(_.snakeCase('HTMLInputElement'))
// 'htmlInputElement'
_.snaceCase('XmlHTTPRequest')
// 'xmlHttpRequest'
```
The examples above look "right" to me. If `_` would separate sequential uppercase characters:

```js
_.snakeCase('HTMLInputElement')
// 'h_t_m_l_input_element'
_.snaceCase('XmlHTTPRequest')
// 'xml_h_t_t_p_request'

_.camelCase(_.snakeCase('HTMLInputElement'))
// 'hTMLInputElement'
_.snaceCase('XmlHTTPRequest')
// 'xmlHTTPRequest'
```

The above cases no longer separate uppercase words. And transforming between snake_case and camelCase _**still**_ do not preserve the correct casing. So trying to preserve casing between transforms is not a reasonable priority IMO.

TL;DR: Looks right to me.

**Comment by shasharoman** [comment]
@csvn The current behavior is not a problem for me. I just don’t understand it. After reading your explanation and examples, I understand that there is no perfect solution here. I agree with you very much. So I close this issue to avoid further discussion.
